games/rummy/scenes/PlayScene.ts

```diff
@@ -0,0 +1,1160 @@
+import { Result } from '@/games/common/types/game';
+import Card from '../../common/Factories/cardImage';
+import Deck from '../../common/Factories/deckImage';
+import GAME from '../../common/constants/game';
+import STYLE from '../../common/constants/style';
+import GamePhase from '../constants/gamePhase';
+import GameResult from '../constants/gameResult';
+import RummyPlayer from '../models/rummyPlayer';
+
+import Table from '../../common/Factories/tableScene';
+import Text = Phaser.GameObjects.Text;
+import Zone = Phaser.GameObjects.Zone;
+import TimeEvent = Phaser.Time.TimerEvent;
+
+export default class PlayScene extends Table {
+  // TODO: private->#に変更する
+  private playerDecks: Array<Deck> = []; // player, house
+
+  private deckSizeTexts: Array<Text> = []; // player, house
+
+  private dropZones: Array<Zone> = []; // player, house
+
+  private dropCardRanks: Array<number> = []; // player, house
+
+  private timeEvent: TimeEvent | undefined;
+
+  private housePlayTimeEvent: TimeEvent | undefined;
+
+  private houseDealTimeEvent: TimeEvent | undefined;
+
+  #countDownSound: Phaser.Sound.BaseSound | undefined;
+
+  private meldZone: Zone | undefined;
+
+  private meldZoneEach: Zone[] = [];
+
+  private meldZoneEachCounter = 0; // 新しいcardが追加されたら、増える。
+
+  private frontDeckZone: Zone | undefined;
+
+  private backDeckZone: Zone | undefined;
+
+  private isPlayerTurn = false; // 初期状態ではハウスのターンとします
+
+  private houseFlag = false;
+
+  private playerFlag = false;
+
+  private deckBack: Deck[] = [];
+
+  private deckFront: Card[] = [];
+
+  private meldZoneSameRank: Card[][] = [[], []];
+
+  private meldZoneConsecutiveRanks: Card[][] = [[], []];
+
+  private arrayCardsPutOnMeldZone: Card[][] = [];
+
+  private meldZoneContainer!: Phaser.GameObjects.Container;
+
+  private turnText: Text | undefined;
+
+  private deckBackCardX = 850;
+
+  private deckBackCardY = 550;
+
+  private deckFrontCardX = 1050;
+
+  private deckFrontCardY = 550;
```

Comment:
deckはcardの集合であることは明らかなので、Cardは冗長化と思います。Cardを削除して、以下のように変更するのはどうでしょうか？

```suggestion
  private deckBackX = 850;

  private deckBackY = 550;

  private deckFrontX = 1050;

  private deckFrontY = 550;
```